src/ixdat/techniques/deconvolution.py

```diff
@@ -10,289 +9,308 @@
+            # find the other parameters from the siq Molecule files
+            Molecule = plugins.siq.Molecule
+            molecule = Molecule.load(mol)
+            if D is None:
+                D = molecule.D
+                print(D)
+            if H_v_cc is None:
+                H_v_cc = 1 / (R * chip.T * molecule.H_0) * 100
+                print(H_v_cc)
+        else:
+            if D is None:
+                raise TechniqueError(
+                    "D is required to initialize ECMSImpluseResponse. Alternatively"
+                    "activate siq as plugin to automatically load D for molecule."
+                )
+            if H_v_cc is None:
+                raise TechniqueError(
+                    "H_v_cc is required to initialize ECMSImpluseResponse. Alternatively"
+                    "activate siq as plugin to automatically load H_v_cc for molecule."
+                )
+        # convert the mol flux to volumetric flux using T and p given by chip.
+        V_dot = n_dot * R * T / p
 
-    def get_limit_frequency():
-        """Calculate the limiting frequency for deconvolution for a given
-        ECMSImpulseResponse.
-        TODO: finish this method"""
+        # make a time array
+        t_kernel = np.arange(0, duration, dt)
+        t_kernel[0] = 1e-6
 
-    def get_limit_time_res():
-        """Calculate the limiting time resolution for deconvolution for a given
-        ImpulseResponse.
-        TODO: finish this method"""
+        tdiff = (
+            t_kernel * D / (working_distance**2)
+        )  # See Krempl et al, 2021. SI: Equ. (6)
 
-    def plot_cond_number_heatmap(cond_number, Z, cc):
-        """
-        TODO: finish this method
+        def calc_H(s):
+            # See Krempl et al, 2021. SI: Equ 24 + definitions for kappa and lamda
+            # (between Equ 11 and 12)
+            kappa = (A_el * 1e-4 * working_distance) / gas_volume
+
+            lambda_ = V_dot / gas_volume * working_distance**2 / D
 
-        Parameters
-        ----------
-        cond_number : TYPE
-            DESCRIPTION.
+            H = 1 / (
+                H_v_cc * (s + lambda_) / kappa * cosh(sqrt(s)) + sqrt(s) * sinh(sqrt(s))
+            )
+            return H
 
-        Returns
-        -------
-        None.
+        # now calculate the modelled impulse response: See Krempl et al, 2021. SI, page
+        # S4: "calculating the discrete values of the impulse response at a sampling
+        # frequency equal to the sampling frequency of the measured mass spectrometer
+        # signal by numerical inverse Laplace transformation with the Talbot method"
+        kernel = np.zeros(len(t_kernel))
+        for i in range(len(t_kernel)):
+            kernel[i] = invertlaplace(calc_H, tdiff[i], method="talbot")
 
-        """
+        if (
+            norm
+        ):  # normalize the kernel intensity to the total area under the ImpulseResponse
+            area = np.trapezoid(kernel, t_kernel)
```

Comment:
This doesn't work with my version of numpy (1.24.3). I had to change it to `trapz` which I think works with all supported versions.

Reply:
see above